lib/es6-promise/-internal.js

```diff
@@ -87,10 +76,7 @@ function handleMaybeThenable(promise, maybeThenable, then) {
       maybeThenable.constructor.resolve === originalResolve) {
     handleOwnThenable(promise, maybeThenable);
   } else {
-    if (then === TRY_CATCH_ERROR) {
-      reject(promise, TRY_CATCH_ERROR.error);
-      TRY_CATCH_ERROR.error = null;
-    } else if (then === undefined) {
+    if (then === undefined) {
```

Comment:
```suggestion
    if (then === void 0) {
```

Reply:
Is there a specific reason to change this? Typically, uglification tools already make this transformation.